06_plot_remove_cancer_results.ipynb

```diff
@@ -0,0 +1,1534 @@
+       "      <td>-0.449780</td>\n",
+       "      <td>7.127790e-08</td>\n",
+       "      <td>0.000030</td>\n",
+       "      <td>True</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>412</th>\n",
+       "      <td>SF3B1_UVM</td>\n",
+       "      <td>-0.792857</td>\n",
+       "      <td>3.991205e-06</td>\n",
+       "      <td>0.000475</td>\n",
+       "      <td>True</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>3</th>\n",
+       "      <td>CDH1_BRCA</td>\n",
+       "      <td>-0.381885</td>\n",
+       "      <td>4.076015e-06</td>\n",
+       "      <td>0.000475</td>\n",
+       "      <td>True</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>136</th>\n",
+       "      <td>PBRM1_KIRC</td>\n",
+       "      <td>-0.422218</td>\n",
+       "      <td>5.060010e-06</td>\n",
+       "      <td>0.000475</td>\n",
+       "      <td>True</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>239</th>\n",
+       "      <td>IDH1_LGG</td>\n",
+       "      <td>-0.196555</td>\n",
+       "      <td>5.690185e-06</td>\n",
+       "      <td>0.000475</td>\n",
+       "      <td>True</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "     identifier  delta_mean       p_value  corr_pval  reject_null\n",
+       "54    BRAF_THCA   -0.449780  7.127790e-08   0.000030         True\n",
+       "412   SF3B1_UVM   -0.792857  3.991205e-06   0.000475         True\n",
+       "3     CDH1_BRCA   -0.381885  4.076015e-06   0.000475         True\n",
+       "136  PBRM1_KIRC   -0.422218  5.060010e-06   0.000475         True\n",
+       "239    IDH1_LGG   -0.196555  5.690185e-06   0.000475         True"
+      ]
+     },
+     "execution_count": 6,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "experiment_comparison_df = au.compare_results(single_cancer_df,\n",
+    "                                              pancancer_df=remove_cancer_df,\n",
+    "                                              identifier='identifier',\n",
+    "                                              metric='aupr',\n",
+    "                                              correction=True,\n",
+    "                                              verbose=False)\n",
+    "experiment_comparison_df.sort_values(by='p_value').head()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/html": [
+       "<div>\n",
+       "<style scoped>\n",
+       "    .dataframe tbody tr th:only-of-type {\n",
+       "        vertical-align: middle;\n",
+       "    }\n",
+       "\n",
+       "    .dataframe tbody tr th {\n",
+       "        vertical-align: top;\n",
+       "    }\n",
+       "\n",
+       "    .dataframe thead th {\n",
+       "        text-align: right;\n",
+       "    }\n",
```

Comment:
Nice plot!

So it looks like you have some genes that very accurately predict mutation status using pancancer but not using pancancer-only. And these genes are not signficantly more accurate compared to your shuffled data. What would you say about these genes? Are these mutations very common in many cancers? I feel like if that were the case they would has also been accurate in pancancer-only.

Is the takeaway that cancer types have very distinct mutational signatures and so removing the cancer type of interest performs very poorly. But then is there an advantage to having some pancancer information?

Reply:
Again, great questions! Thanks for diving deep into this, it really helps.

> So it looks like you have some genes that very accurately predict mutation status using pancancer but not using pancancer-only. And these genes are not signficantly more accurate compared to your shuffled data. What would you say about these genes? Are these mutations very common in many cancers? I feel like if that were the case they would has also been accurate in pancancer-only.

I don't know if there's anything especially significant about the genes that show a strong difference in performance without the data from the relevant cancer type. I only ran one replicate of 4-fold CV here (for the experiments in the other notebook I ran 2x4-fold CV), so I suspect that with an additional replicate we'd see a lot more significant genes on the left of these plots.

Mostly I was interested in the general trend (to the left or roughly equal), but that appears to be behaving the way we expected (shifted to the left in these volcano plots) so I think I'll leave it at that for now and move onto other things.

> Is the takeaway that cancer types have very distinct mutational signatures and so removing the cancer type of interest performs very poorly. But then is there an advantage to having some pancancer information?

Yeah, basically - this experiment was mainly a sanity check to see that in most cases, removing the cancer type we're testing on from the training set hurts performance. The volcano plots show pretty clearly that this is generally the case, so that makes us more confident that things are running correctly.

To answer your second question, I would say yes, in general there is usually still an advantage to having pan-cancer information, even if it's not the right cancer type. See, for instance, this example from the notebook:

![image](https://user-images.githubusercontent.com/2345877/95599498-2838b100-0a1f-11eb-8f4c-519437aa66ef.png)

If you look at the `pancancer_only` box plot, you can see that the model clearly outperforms the shuffled baseline, even without data from the relevant cancer type. So there does seem to be some signal in data from other cancer types, at least in the case of BRAF (and for most of the other genes I've checked by hand).

The best way to use this in a pan-cancer model, however, is unclear. See this plot from `05_plot_cancer_type_results.ipynb`:

![image](https://user-images.githubusercontent.com/2345877/95599804-90879280-0a1f-11eb-8a83-e015f912ebc9.png)

Gene/cancer type combinations on the right have better performance when pan-cancer data is used. There are very few examples that significantly outperform the single-cancer models, although there are a small handful.

Taken together, these results suggest that there is useful signal in the data for unrelated cancer types, but maybe we're not using it in the right way with our simple linear model. It's possible that better models or different approaches to incorporating data from unrelated cancers would help to shift more points on this plot to the right. This is something Casey and I have been thinking about and we have a few ideas, so stay tuned.